xwiki-platform-core/xwiki-platform-appwithinminutes/xwiki-platform-appwithinminutes-test/xwiki-platform-appwithinminutes-test-pageobjects/src/main/java/org/xwiki/appwithinminutes/test/po/ApplicationClassEditPage.java

```diff
@@ -97,14 +96,15 @@ public boolean hasPreviousStep()
      * Drags a field of the specified type from the field palette to the field canvas.
      *
      * @param fieldType the type of field to add, as displayed on the field palette
+     * @return the newly added field page object
      */
     public ClassFieldEditPane addField(String fieldType)
     {
         String fieldXPath = "//li[@class = 'field' and normalize-space(.) = '%s']";
         WebElement field = palette.findElement(By.xpath(String.format(fieldXPath, fieldType)));
         // NOTE: We scroll the page up because the drag&drop fails sometimes if the dragged field and the canvas (drop
         // target) are not fully visible. See https://code.google.com/p/selenium/issues/detail?id=3075 .
-        getDriver().createActions().sendKeys(Keys.HOME).perform();
+        getDriver().executeScript("window.scrollTo(0, 0)");
```

Comment:
Might be interesting to move this to a `moveTop` method in `BaseElement`.
WDYT?

Reply:
Well actually at the very least you should use `getDriver().scrollTo()` method here. Not sure if we need it in BaseElement now: that's a pretty low-level API so it makes sense if it reminds in XWikiWebDriver IMO.

Reply:
`XWikiWebDriver#scrollTo` takes a WebElement, not coordinates.
But we could introduce a `XWikiWebDriver#scrollTo(int xCoord, int yCoord)` method.

Reply:
Moving to `(0, 0)` is low level. Moving to the top of a page is more business related imo.

Reply:
> XWikiWebDriver#scrollTo takes a WebElement, not coordinates.

Right, read too fast. Would be a `scrollToTop` method then. +1 to add that: the test would still be fragile but it was already and at least it's easier to spot where this hack is used if we ever find a way to improve it (which I doubt since the drag&drop can be difficult even manually sometimes...).